web/cobrands/fixmystreet/fixmystreet.js

```diff
@@ -1218,10 +1247,19 @@ fixmystreet.display = {
             // wrong (but what is shown is correct).
             $('.js-back-to-report-list').attr('href', fixmystreet.original.href);
 
-            // Problems nearby should act the same as 'Back to all reports' on around,
-            // but on /my and /reports should go to that around page.
-            if (fixmystreet.original.page == 'around') {
-                $sideReport.find('#key-tool-problems-nearby').addClass('js-back-to-report-list');
+            // Problems nearby on /my should go to the around page,
+            // otherwise show reports within the current map view.
+            if (fixmystreet.original.page === 'around' || fixmystreet.original.page === 'reports') {
+                $sideReport.find('#key-tool-problems-nearby').click(function(e) {
+                    fixmystreet.display.reports_list(fixmystreet.original.href, function() {
```

Comment:
Interestingly, going direct to a report page, the "Problems nearby" link becomes 'live' and updates as you pan the map, then taking you to where you are. This doesn't otherwise happen, but perhaps it should, I imagine that would be nice to have here too (but later).